src/server.cpp

```diff
@@ -44,6 +46,22 @@ server::server(const std::string& address, const std::string& port)
   start_accept();
 }
 
+bool server::isValid(const std::string& address, const std::string& port){
```

Comment:
Does this ever get called?

I think it makes sense to have these checks here instead of in `main.cpp` but we should probably delete where this code occurs `main.cpp` and just call isValid in the server's constructor